src/main/java/com/direwolf20/buildinggadgets/client/gui/materiallist/MaterialListGUI.java

```diff
@@ -1,217 +1,250 @@
+        Preconditions.checkArgument(item.getCapability(CapabilityTemplate.TEMPLATE_KEY_CAPABILITY).isPresent());
+        this.item = item;
+    }
+
+    @Override
+    public void init() {
+        this.backgroundX = getXForAlignedCenter(0, width, BACKGROUND_WIDTH);
+        this.backgroundY = getYForAlignedCenter(0, height, BACKGROUND_HEIGHT);
+
+        TemplateHeader header = getTemplateHeader();
+        String name = header.getName();
+        String author = header.getAuthor();
+        this.title = name == null && author == null ? MaterialListTranslation.TITLE_EMPTY.format()
+                : name == null ? MaterialListTranslation.TITLE_AUTHOR_ONLY.format(author)
+                : author == null ? MaterialListTranslation.TITLE_NAME_ONLY.format(name)
+                : MaterialListTranslation.TITLE.format(name, author);
+        this.titleTop = getYForAlignedCenter(backgroundY, getWindowTopY() + ScrollingMaterialList.TOP, font.FONT_HEIGHT);
+        this.titleLeft = getXForAlignedCenter(backgroundX, getWindowRightX(), font.getStringWidth(title));
+
+        this.scrollingList = new ScrollingMaterialList(this);
+        // Make it receive mouse scroll events, so that the player can use his mouse wheel at the start
+        this.setFocused(scrollingList);
+        this.children.add(scrollingList);
+
+        int buttonY = getWindowBottomY() - (ScrollingMaterialList.BOTTOM / 2 + BUTTON_HEIGHT / 2);
+        this.buttonClose = new Button(0, buttonY, 0, BUTTON_HEIGHT, MaterialListTranslation.BUTTON_CLOSE.format(), b -> Minecraft.getInstance().player.closeScreen());
+        this.buttonSortingModes = new Button(0, buttonY, 0, BUTTON_HEIGHT, scrollingList.getSortingMode().getLocalizedName(), (button) -> {
+            scrollingList.setSortingMode(scrollingList.getSortingMode().next());
+            buttonSortingModes.setMessage(scrollingList.getSortingMode().getLocalizedName());
+        });
+        this.buttonCopyList = new Button(0, buttonY, 0, BUTTON_HEIGHT, MaterialListTranslation.BUTTON_COPY.format(), (button) -> {
+            getMinecraft().keyboardListener.setClipboardString(stringify(Screen.hasControlDown()));
+            getMinecraft().player.sendStatusMessage(new TranslationTextComponent(MaterialListTranslation.MESSAGE_COPY_SUCCESS.getTranslationKey()), true);
+        });
+
+        // Buttons will be placed left to right in this order
+        this.addButton(buttonSortingModes);
+        this.addButton(buttonCopyList);
+        this.addButton(buttonClose);
+        this.children.add(scrollingList);
+        this.calculateButtonsWidthAndX();
+    }
+
+    private String stringify(boolean detailed) {
+        if (detailed)
+            return stringifyDetailed();
+        return stringifySimple();
+    }
+
+    private String stringifyDetailed() {
+        MaterialList materialList = getTemplateHeader().getRequiredItems();
+        return materialList == null ? "" : materialList.stringifySimple();
```

Comment:
I'm not 100% sure on why it wouldn't have a material list?

Reply:
Because it might not be cached? A MaterialList can only be evaluated in in IBuildView (and possibly in a Transaction) as that is the only place with the apropiate context for this calculation. Even if we'd pass the context down (which would add an unnecessary parameter to this, which we might have to makw nullable) to the creation of the TemplateHeader it might not be possible to create a BuildView or if we enforce it we might stop running transactions - if you want to we can add a utility/ default Method which does exactly this, but it's still valid for it not to know the materials in my opinion